cpp/src/KIM_SimulatorModelImplementation.cpp

```diff
@@ -847,6 +847,8 @@ int SimulatorModelImplementation::Initialize(
     LOG_DEBUG("Exit 1=" + callString);
     return true;
   }
+  // Convert path from string representation.
+  parameterFileDirectoryName_ = parameterFileDirectoryNameString_;
```

Comment:
That's only there, I think, because I was shying away from changing the signature of `SharedLibrary::GetParameterFileDirectoryName()`, which currently accepts only `std::string*` and not `FILESYSTEM::Path*`. But I now realized that `SharedLibrary::GetParameterFileDirectoryName()` is not a public API (in contrast to `SimulatorModel::GetParameterFileDirectoryName()`) and we thus do not need this round-trip conversion between string and path representation. I am going to rework this part.

Reply:
We should just remove these string variables.

Reply:
Good